### ISSUE TITLE
"zoomed screen" when taking screenshot after disconnecting from dual monitor

### ISSUE THREAD

**Issue by asafgaon** [flameshot-org/flameshot#3093]
[](url)I work with my laptop (MBP) in two modes, either on the laptop or connected to 2 external monitors (via dock). When I take screenshots using Flameshot with the two monitors everything is good, but after I disconnect the external monitors and return to to work on the laptop, every time I try to take a screenshot I can only see what can be described as a "zoomed screen" to screenshot (see screenshot). The only way to solve it is to exit Flameshot and start it again.

Ways to reproduce: connect MBP to dual screens, take a screenshot successfully, disconnect MBP, try to take a screenshot, see "zoomed screen"

Version: Flameshot v12.1.0 (96c2c82e) Compiled with Qt 5.15.5
Platform: Macbook Pro (2009 13") running MacOS 13.2 
![zoomed screen](https://user-images.githubusercontent.com/69471369/217355741-b834314b-7fc2-4d1b-af87-e6761feedf45.png)

**Comment by wirew0lf** [comment]
I'm having the same issue

**Comment by jeremydescamps** [comment]
Same here !

Here is my information: 
- Flameshot v12.1.0 (96c2c82e)
- Compiled with Qt 5.15.5
- darwin: 22.3.0
- osx: 13.2

**Comment by pg42819** [comment]
ever resolved? I'm getting the same months later

**Comment by Blackthron** [comment]
Not resolved. Having same issue. This stops me from using the software comepletely. It's weird that devs are not fixing in.

**Comment by Xat59** [comment]
Yeah I confirm it is still not fixed

**Comment by faagner7** [comment]
Same issue

**Comment by hainguyen1107** [comment]
Same issue!

**Comment by hammad-Ikhlaq-7vals** [comment]
Same issue

**Comment by automagiq** [comment]
same issue : Flameshot zooming in on my Mac when I try to take a screenshot, I use two monitors but it does it even when disconnect another monitor. 

MacOs Sonoma 14.0 
Flameshot v12.1.0 (96c2c82e)
Compiled with Qt 5.15.5
darwin: 23.0.0
osx: 14.0

![image](https://github.com/flameshot-org/flameshot/assets/49632874/7eec56fc-cffc-47a7-86cb-6210afe3483e)

**Comment by mischavandenburg** [comment]
Same here

**Comment by mrkrstphr** [comment]
I'm having this issue, but it does seem to work correctly again if I quit Flameshot and re-open it.

**Comment by frohro** [comment]
I'm having this issue too.  I'm not sure if I connected this to dual monitors though.  When I quit flameshot and restart it, I get the message that there was a flameshot error and I can't take a screenshot.  This is on Wayland, Ubuntu 23.10.

**Comment by AndreasGrip** [comment]
Just got the same problem and it was resolved by quit flameshot and starting it again..
Macbook pro 16" 2023.
Usually connected to one external 4k monitor and one 1920x1200, but when problem occured I was traveling and only built in screen used.

**Comment by SghaierFiras** [comment]
Same solution said by @AndreasGrip worked for me too.
Quit Flameshot and restart it again. everything worked fine.

**Comment by JasCodes** [comment]
@SghaierFiras Thats not solution to restart everytime I restart the machine

**Comment by 0xGodson** [comment]
Same issue.

**Comment by ousiadigital** [comment]
same here

**Comment by AlwaysNoobCoder** [comment]
same, but for me it fedora linux 37, and restart app not works too.

**Comment by iamludal** [comment]
Same here 😞

**Comment by apasias** [comment]
Same

**Comment by JasCodes** [comment]
Kinna major bug...

**Comment by adrianmusante** [comment]
> looks like duplicate of #2612 Also as suggested [here](https://github.com/flameshot-org/flameshot/issues/3093#issuecomment-1907490885) and [there](https://github.com/flameshot-org/flameshot/issues/2612#issuecomment-1131678178) - issue seems to happen only after monitor setup change (connect/disconnect monitor). So, as workaround - restart the flameshot in this case
> 
> as a bit "better" automated workaround - one can use https://support.apple.com/en-gb/guide/automator/welcome/mac to restart flameshot on monitor connect/disconnect

Could you share the AppleScript or Automator workflow you used to relaunch the application? Thanks in advance.

**Comment by hambro** [comment]
I added a signal handler to `yabai`, listening to display added/removed and restarting flameshot. 
```
yabai -m signal --add event=display_added action="pkill flameshot && sleep 1 && open -n /Applications/flameshot.app"
yabai -m signal --add event=display_removed action="pkill flameshot && sleep 1 && open -n /Applications/flameshot.app"
```
